enumerables.rb

```diff
@@ -0,0 +1,101 @@
+module Enumerable
+
+    def my_each
+        for i in 0..self.length-1
+            yield(self[i])
+        end
+    end
+    
+    def my_each_with_index
+        for i in 0..self.length-1
+            yield(self[i],i)
+        end
+    end
+
+    def my_select
+        arr = []
+        for i in 1..self.length
+            if yield(i)
+                arr.push(i)
+            end
+        end
+        arr
+    end
```

Comment:
* [ ] my_select returns an array containing all elements of enum for which the given block returns a true value
  ```
    block = proc {|x| x>4}
    array = [1,2,5,8,7,98,4]
    p array.my_select(&block) == array.select(&block) # false
  ```
* [ ] my_select returns an enumerator if no block is given
  ```
    array = [1,2,5,8,7,98,4]
    p array.my_select.class # not Enumerator
  ```